src/components/ReactPortal.js/ReactPortal.js

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useRef } from "react";
+import { createPortal } from "react-dom";
+
+const modalRoot = document.getElementById("modal");
+
+export function ReactPortal({ children }) {
+  const elRef = useRef(null);
+
+  if (!elRef.current) {
+    elRef.current = document.createElement("div");
+  }
+
+  useEffect(() => {
+    modalRoot.appendChild(elRef.current);
+
+    return () => modalRoot.removeChild(elRef.current);
+  });
+
+  return createPortal(children, elRef.current);
```

Comment:
why not return jsx from create portal function ? , Not sure if using vanilla js here is ideal.

Reply:
It is a general portal function, we can pass any children props to it and it will get rendered inside the target element here `elRef.current` 

Vanilla JS is used because we are accessing the elements outside of the `root`